### PR TITLE
Add FunctionBuilder

### PR DESCRIPTION
**Pull request by bnoordhuis**
v8::Function and v8::FunctionTemplate have a lot of options that can be
configured. The builder() associated functions on those types help with
that.

<hr>

cc @ry - this is more groundwork for fast API calls.

cc @devsnek - I'll concede that I was wrong and you were right. 🙂

### CODE REVIEW

**Comment by devsnek** [comment]
I can update https://github.com/denoland/rusty_v8/pull/454 if you want.

**Comment by bnoordhuis** [comment]
Oh, I opened this PR because I didn't have the heart to ask you to fix the conflicts in #454.

File: src/function.rs [hunk 1]
```diff
@@ -63,6 +67,38 @@ extern "C" {
...
+#[repr(C)]
+pub enum SideEffectType {
+  HasSideEffect,
+  HasNoSideEffect,
+  HasSideEffectToReceiver,
+}
+
+#[repr(C)]
+#[derive(Default)]
+pub(crate) struct CFunction([usize; 2]);
```
Comment (ry) [inline src/function.rs:100]: Where does the value 2 come from? I tried adding a size assert for this and it failed.

Comment (ry) [reply src/function.rs:100]: Nevermind - I didn't have the include for v8-fast-api-calls.h

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**